cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Transform.java

```diff
@@ -62,7 +62,10 @@ public void configure(StageConfigurer configurer) {
   }
 
   /**
-   * Initialize the Transform Stage. Called during the runtime with context of the Transform.
+   * Initialize the Transform Stage. Transforms are initialized once when the program starts up. Initialization
+   * is guaranteed to occur before any calls to {@link #transform(Object, Emitter)} are made. Transforms are also
+   * initialized once during pipeline creation, so property validation should be done here in order to fail creation
+   * of pipelines with invalid config.
```

Comment:
hm i guess we can have a separate method. but im going to remove transform related stuff from this PR and we'll do it separately.